frontend/front/src/pages/Surveyor/account/edit/EditAccount.js

```diff
@@ -38,168 +61,195 @@ const EditAccount = () => {
-            variant="standard"
-            defaultValue={address}
-            {...register("address", {
-              required: {
-                value: true,
-                message: "Please Enter Address",
-              },
-            })}
-          />
-          <span style={errorStyles}>{errors?.address?.message}</span>
-          <TextField
-            id="standard-basic"
-            placeholder="Enter Phone Number"
-            type="tel"
-            style={btnstyle}
-            name="phoneNumber"
-            fullWidth
-            label="Phone Number"
-            variant="standard"
-            defaultValue={phoneNumber}
-            {...register("phoneNumber", {
-              required: {
-                value: true,
-                message: "Please Enter Phone Number",
-              },
-              maxLength: {
-                value: 11,
-                message: "Phone number too long",
-              },
-              minLength: {
-                value: 8,
-                message: "Phone number too short",
-              },
-            })}
-          />
-          <span style={errorStyles}>{errors?.phoneNumber?.message}</span>
+          <form onSubmit={handleSubmit(EditAccountForms)}>
+            <TextField
+              id="standard-basic"
+              placeholder="Enter First Name"
+              type="text"
+              style={btnstyle}
+              name="firstName"
+              fullWidth
+              label="First Name"
+              variant="standard"
+              defaultValue={firstName}
+              {...register("firstName", {
+                required: {
+                  value: true,
+                  message: "Please Enter First Name",
+                },
+              })}
+            />
+            <span style={errorStyles}>{errors?.firstName?.message}</span>
+            <TextField
+              id="standard-basic"
+              placeholder="Enter Last Name"
+              type="text"
+              style={btnstyle}
+              name="lastName"
+              fullWidth
+              label="Last Name"
+              variant="standard"
+              defaultValue={lastName}
+              {...register("lastName", {
+                required: {
+                  value: true,
+                  message: "Please Enter Last Name",
+                },
+              })}
+            />
+            <span style={errorStyles}>{errors?.lastName?.message}</span>
+            <TextField
+              id="standard-basic"
+              placeholder="Enter Email"
+              type="email"
+              style={btnstyle}
+              name="email"
+              fullWidth
+              label="Email"
+              variant="standard"
+              defaultValue={email}
+              {...register("email", {
+                required: {
+                  value: true,
+                  message: "Please Enter Email",
+                },
+                pattern: {
```

Comment:
For all of these but especially the email, you might be able to be more concise by reusing the components for the SurveyComponent, HeatPumpTextField etc.

Reply:
With HeatPumpTextField, I don't see that I can give it a `type="email"` attribute, or some of the other props like the `pattern` for regex testing. Maybe give it a `textFieldProps` prop or something.